addons/sale/models/sale_order.py

```diff
@@ -589,9 +589,11 @@ def _compute_partner_credit_warning(self):
             show_warning = order.state in ('draft', 'sent') and \
                            order.company_id.account_use_credit_limit
             if show_warning:
-                updated_credit = order.partner_id.commercial_partner_id.credit + (order.amount_total / order.currency_rate)
+                order_sudo = order.sudo()
                 order.partner_credit_warning = self.env['account.move']._build_credit_warning_message(
-                    order, updated_credit)
+                    order_sudo,
+                    order_sudo.partner_id.credit + order.amount_total / order.currency_rate,
```

Comment:
```suggestion
                    order_sudo.partner_id.commercial_partner_id.credit + order.amount_total / order.currency_rate,
```
why did you remove the `.commercial_partner_id` ?

Reply:
good question, no idea, re-added it